service/health/key/provider.go

```diff
@@ -0,0 +1,14 @@
+package key
+
+// Providers is an array of providers which are currently implemented for health checking
+var Providers = [...]string{"aws", "azure", "kvm"}
+
+// IsKnownProvider is a helper function returning true iff the given value is in Providers
```

Comment:
Is this a typo ?

```suggestion
// IsKnownProvider is a helper function returning true if the given value is in Providers
```

Reply:
It's something I picked up in my math classes :) `iff` is short for `if and only if`. It's the biconditional meaning `true` will be returned if the value is in `Providers` *and* `false` if the given value is not in `Providers`. This isn't implied if we only use `if`.